tools/SourceKit/lib/SwiftLang/SwiftEditor.cpp

```diff
@@ -963,6 +963,17 @@ class SemanticAnnotator : public SourceEntityWalker {
     if (AvailableAttr::isUnavailable(D))
       return true;
 
+    auto &SM = D->getASTContext().SourceMgr;
+    if (D == D->getASTContext().getOptionalNoneDecl() &&
+        SM.extractText(Range) == "nil") {
+      // If a 'nil' literal occurs in a swift-case statement, it gets replaced
+      // by a reference to 'Optional.none' in the AST. We want to continue
+      // highlighting 'nil' as a keyword and not as an enum element.
+      // Detect this case if 'none' was spelled with 3 characters, which
+      // indicates that it was rewritten from 'nil'.
```

Comment:
This description is not correct anymore :)

Reply:
Thanks for catching it. I removed the last sentence.